### PR TITLE
Add failure scenario integration tests for empty gcs file for streaming writes.

### PR DESCRIPTION
**Pull request by meet2mky**
### Description
This PR extends the existing failure tests for local files to also cover scenarios with empty GCS files.

* Introduced a validator interface to check files from Google Cloud Storage (GCS) for both empty GCS files and newly created local files.
* Fixed the customEndpoint format as Bucket.Attr call was failing when creating empty gcs file because it expected the endpoint in the `http://localhost:8020/storage/v1/` format.
* Test directory cleanup is also not working on the TestBench server. To resolve this, we create randomized test directory path (testDirPath) for each test (b/397575791).

### Link to the issue in case of a bug fix.
b/388720840

### Testing details
1. Manual - Done
2. Unit tests - NA
3. Integration tests - Done

### CODE REVIEW

File: tools/integration_tests/emulator_tests/streaming_writes_failure/new_local_file_failure_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	var err error
+	t.data, err = operations.GenerateRandomData(5 * operations.MiB)
+	require.NoError(t.T(), err)
+	log.Printf("Running tests with flags for new local files: %v", t.flags)
+}
+
+func (t *newLocalFileFailureTestSuite) SetupTest() {
+	t.setupTest()
+	// Create local file.
+	t.filePath, t.fh1 = CreateLocalFileInTestDir(t.ctx, t.storageClient, t.testDirPath, FileName1, t.T())
```
Comment (ashmeenkaur) [inline tools/integration_tests/emulator_tests/streaming_writes_failure/new_local_file_failure_test.go:52]: Can we move `ValidateObjectNotFoundErrOnGCS()` here and remove the interface and test usage of `validateGcsObject()` method for simplicity.

File: tools/integration_tests/emulator_tests/streaming_writes_failure/empty_gcs_file_failure_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+type emptyGcsFileFailureTestSuite struct {
+	commonFailureTestSuite
+}
+
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func (t *emptyGcsFileFailureTestSuite) SetupSuite() {
```
Comment (ashmeenkaur) [inline tools/integration_tests/emulator_tests/streaming_writes_failure/empty_gcs_file_failure_test.go:40]: SetupSuite function is identical for both the local and empty GCS test suites. Could we consolidate this into the common SetupSuite function in the shared commonFailureTestSuite, and set t.configPath within the SetupTest functions of the respective wrapper suites?

File: tools/integration_tests/emulator_tests/streaming_writes_failure/new_local_file_failure_test.go (hunk 1 above)
Comment (meet2mky) [reply tools/integration_tests/emulator_tests/streaming_writes_failure/new_local_file_failure_test.go:52]: `func CreateLocalFileInTestDir` already does this validation `ValidateObjectNotFoundErrOnGCS()`. The interface and `validateGcsObject()` is to validate the object from gcs in the `assert` part of the tests after the failure from emulator is introduced in both the cases when the object should be of empty size and when it shouldn't be present.

Comment (ashmeenkaur) [reply tools/integration_tests/emulator_tests/streaming_writes_failure/new_local_file_failure_test.go:52]: sorry my bad, didn't realize that this was being used in a helper function. LGTM

File: tools/integration_tests/emulator_tests/streaming_writes_failure/empty_gcs_file_failure_test.go (hunk 2 above)
Comment (meet2mky) [reply tools/integration_tests/emulator_tests/streaming_writes_failure/empty_gcs_file_failure_test.go]: Done!

**Review by ashmeenkaur — APPROVED** [review]
